drivers/pcie/host/pcie.c

```diff
@@ -68,11 +68,78 @@ uintptr_t pcie_get_mbar(pcie_bdf_t bdf, unsigned int index)
 	return PCIE_CONF_BAR_ADDR(addr);
 }
 
+/* The first bit is used to indicate whether the list of reserved interrupts
+ * have been initialized based on content stored in the irq_alloc linker
+ * section in ROM.
+ */
+#define IRQ_LIST_INITIALIZED 0
+
+static ATOMIC_DEFINE(irq_reserved, CONFIG_MAX_IRQ_LINES);
+
+static unsigned int irq_alloc(void)
+{
+	int i;
+
+	for (i = 0; i < ARRAY_SIZE(irq_reserved); i++) {
+		unsigned int fz;
+
+		while ((fz = find_lsb_set(~atomic_get(&irq_reserved[i])))) {
+			fz--; /* find_lsb_set returns index + 1 */
+
+			if (!atomic_test_and_set_bit(&irq_reserved[i], fz)) {
+				return fz + (i * sizeof(atomic_val_t) * 8);
```

Comment:
Might want to make sure the result is less than `CONFIG_MAX_IRQ_LINES`. Or else, this might return 5 when `CONFIG_MAX_IRQ_LINES==4`, since `find_lsb_set()` sees bit 5 as set in the above while condition (assuming atomic_val_t is at least an `uint8_t`).

Reply:
That's a good point, the way `irq_reserved` is defined it will fit bits rounded up to the next multiple of `(sizeof(atomic_val_t) * 8`, which might be more than `MAX_IRQ_LINES`. I'll fix it.